sphinx_gallery/gen_gallery.py

```diff
@@ -254,6 +255,20 @@ def call_memory(func):
     if (not isinstance(last_cell, str)) and (last_cell is not None):
         raise ConfigError("The 'last_notebook_cell' parameter must be type str"
                           " or None, found type %s" % type(last_cell))
+    # Check pypandoc
+    if gallery_conf['pypandoc']:
+        pypandoc = gallery_conf['pypandoc']
+        if not isinstance(pypandoc, (bool, dict)):
+            raise ConfigError("'pypandoc' must be a dict or bool, got: "
+                              "%s" % type(pypandoc))
+        if isinstance(pypandoc, dict):
+            accepted_keys = ('extra_args', 'filters')
+            for key in pypandoc.keys():
+                if key not in accepted_keys:
+                    raise ConfigError("'pypandoc' only accepts the following "
+                                      "key values: %s, got: %s."
+                                      % accepted_keys, key)
```

Comment:
... here you do the triaging to make `pypandoc` False or `dict` (only allowing the latter if it's avaliable). So something like:
```
pypandoc = dict() if pypandoc is True else pypandoc
if isinstance(pypandoc, dict):
    if not _has_pypandoc():
        warn(...)
        pypandoc = False
    else:
        accepted_keys ...
        ...
gallery_conf['pypandoc'] = pypandoc  # now either False or a dict of options to pass
```

Reply:
Thanks that's nicer!